Meta/gn/secondary/Kernel/BUILD.gn

```diff
@@ -0,0 +1,849 @@
+      "Arch/aarch64/RPi/UART.cpp",
+      "Arch/aarch64/RPi/Watchdog.cpp",
+      "Arch/aarch64/SafeMem.cpp",
+      "Arch/aarch64/SmapDisabler.cpp",
+      "Arch/aarch64/TrapFrame.cpp",
+      "Arch/aarch64/boot.S",
+      "Arch/aarch64/kprintf.cpp",
+      "Arch/aarch64/vector_table.S",
+    ]
+  }
+}
+
+source_set("lib_sources") {
+  deps = [
+    ":kernel_debug_gen",
+    "//Userland/Libraries/LibC:install_libc_headers",
+    "//Userland/Libraries/LibVT:generate_vt_state_machine(//Meta/gn/build/toolchain:serenity)",
+  ]
+
+  configs += [ ":Kernel_config" ]
+  sources = [
+    "//AK/DOSPackedTime.cpp",
+    "//AK/Error.cpp",
+    "//AK/Format.cpp",
+    "//AK/GenericLexer.cpp",
+    "//AK/Hex.cpp",
+    "//AK/MemoryStream.cpp",
+    "//AK/Stream.cpp",
+    "//AK/StringBuilder.cpp",
+    "//AK/StringUtils.cpp",
+    "//AK/StringView.cpp",
+    "//AK/Time.cpp",
+    "//AK/UUID.cpp",
+    "//Userland/Libraries/LibCrypto/Cipher/AES.cpp",
+    "//Userland/Libraries/LibCrypto/Hash/SHA2.cpp",
+    "//Userland/Libraries/LibEDID/DMT.cpp",
+    "//Userland/Libraries/LibEDID/EDID.cpp",
+    "//Userland/Libraries/LibEDID/VIC.cpp",
+    "//Userland/Libraries/LibELF/Image.cpp",
+    "//Userland/Libraries/LibELF/Validation.cpp",
+    "//Userland/Libraries/LibPartition/DiskPartitionMetadata.cpp",
+    "//Userland/Libraries/LibPartition/EBRPartitionTable.cpp",
+    "//Userland/Libraries/LibPartition/GUIDPartitionTable.cpp",
+    "//Userland/Libraries/LibPartition/MBRPartitionTable.cpp",
+    "//Userland/Libraries/LibPartition/PartitionTable.cpp",
+    "//Userland/Libraries/LibPartition/PartitionableDevice.cpp",
+    "//Userland/Libraries/LibVT/EscapeSequenceParser.cpp",
+    "//Userland/Libraries/LibVT/Line.cpp",
+    "//Userland/Libraries/LibVT/Terminal.cpp",
+  ]
+}
+
+action("generate_version_header") {
+  script = "//Meta/gn/secondary/Kernel/generate_version_header.py"
+  outputs = [ "$target_gen_dir/Version.h" ]
+  args = [ rebase_path(outputs[0], root_build_dir) ]
+}
+
+if (current_cpu == "x86_64") {
+  action("preprocess_linker_script") {
+    script = "//Meta/gn/secondary/Ladybird/invoke_process_with_args.py"
+    inputs = [ "Arch/x86_64/linker.ld" ]
+    outputs = [ "$target_gen_dir/linker.ld" ]
+    target_string = ""
+    if (serenity_toolchain == "Clang") {
+      target_string = "--target=x86_64-pc-serenity"
+    }
+    args = [
+      "$serenity_cxx",
+      "$target_string",
+      "-E",
+      "-P",
+      "-x",
+      "c",
+      rebase_path(inputs[0], root_build_dir),
+      "-o",
+      rebase_path(outputs[0], root_build_dir),
+    ]
+  }
+}
+
+executable("Kernel_bin") {
+  configs += [ ":Kernel_config" ]
+  deps = [
+    ":arch_sources",
+    ":generate_version_header",
+    ":kernel_debug_gen",
+    ":kernel_heap",
+    ":lib_sources",
+    ":mini_stdlib_sources",
+    "//Userland/Libraries/LibC:install_libc_headers",
+  ]
+  if (serenity_arch == "x86_64") {
+    deps += [ ":preprocess_linker_script" ]
+  }
+  sources = [
```

Comment:
This source list is massive; I'll move the sources into subdirectories in a follow-up clean up pass after more things are working. At the moment this simply mirrors the CMakeLists.txt